### PR TITLE
Alex branch

### PR DESCRIPTION
**Pull request by Calvin906**
Created Homepage and added two fragments to it. I had to put off this for a bit to get school under control. I will be working more steadily from now on.

### CODE REVIEW

File: app/src/main/java/preston/com/stockapp/HomeActivity.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.home_activity);
+
+        welcome = (TextView) findViewById(R.id.welcome_home);
+        signUp = (TextView) findViewById(R.id.sign_up_home);
+        login = (TextView) findViewById(R.id.login_home);
+
+        Typeface face = Typeface.createFromAsset(getAssets(), "bondini.ttf");
```
Comment (JPrez38) [inline app/src/main/java/preston/com/stockapp/HomeActivity.java]: this is good. I can teach you a trick that you can set the typeface in xml. You basically just have an override on a general activity. But it's a great start. Nice job.

File: app/src/main/java/preston/com/stockapp/HomeActivity.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        welcome.setTypeface(face);
+        signUp.setTypeface(face);
+        login.setTypeface(face);
+
+        signUp.setOnClickListener(new View.OnClickListener() {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/stockapp/HomeActivity.java]: Just implement the onClickListener on the activity. Use a switch(v.getId) to distinguish the specific views

**Review by JPrez38 — CHANGES_REQUESTED** [review]

File: app/src/main/java/preston/com/stockapp/util/LoginFragment.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.login_view, container);
+        TextView cancel = (TextView) view.findViewById(R.id.cancel_button_login);
+
+        cancel.setOnClickListener(new View.OnClickListener() {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/stockapp/util/LoginFragment.java]: Again implement onClickListner in the class

File: app/src/main/java/preston/com/stockapp/util/SignUpFragment.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import preston.com.stockapp.R;
+
+/**
+ * Created by Alex Preston on 9/20/16.
+ */
+
+public class SignUpFragment extends DialogFragment {
+
+
+    public SignUpFragment() {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/stockapp/util/SignUpFragment.java]: Shouldn't have your own constructor in fragments

File: app/src/main/java/preston/com/stockapp/util/SignUpFragment.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.sign_up_view, container);
+        TextView cancel = (TextView) view.findViewById(R.id.cancel_button_sign_up);
+
+        cancel.setOnClickListener(new View.OnClickListener() {
```
Comment (JPrez38) [inline app/src/main/java/preston/com/stockapp/util/SignUpFragment.java]: implement listener

File: app/src/main/res/layout/home_activity.xml [hunk 6]
```diff
@@ -3,6 +3,44 @@
...
+    <TextView
+        android:id="@id/welcome_home"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerHorizontal="true"
+        android:layout_marginTop="@dimen/vertical_start"
+        android:text="@string/welcome_home"
+        android:textSize="@dimen/welcome_text_size"/>
+
+    <RelativeLayout
```
Comment (JPrez38) [inline app/src/main/res/layout/home_activity.xml:15]: whole thing could probably be one flat relative layout

File: app/src/main/res/layout/login_view.xml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    <TextView
+        android:id="@+id/login_text_frag"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentTop="true"
+        android:layout_centerHorizontal="true"
+        android:text="@string/login"
+        android:textSize="@dimen/Medium_text_size"/>
+
+    <RelativeLayout
```
Comment (JPrez38) [inline app/src/main/res/layout/login_view.xml:15]: you don't need nested relative layouts.

File: app/src/main/res/layout/login_view.xml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+            android:layout_height="wrap_content"
+            android:layout_below="@+id/email_login"
+            android:hint="@string/password"
+            android:inputType="textPassword"
+            android:maxLength="25"
+            android:maxLines="1"/>
+
+    </RelativeLayout>
+
+    <RelativeLayout
```
Comment (JPrez38) [inline app/src/main/res/layout/login_view.xml:47]: another nested relative layout. It's slow performance.

File: app/src/main/res/layout/sign_up_view.xml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    <TextView
+        android:id="@+id/text_sign_up"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentTop="true"
+        android:layout_centerHorizontal="true"
+        android:text="@string/sign_up"
+        android:textSize="@dimen/Medium_text_size"/>
+
+    <RelativeLayout
```
Comment (JPrez38) [inline app/src/main/res/layout/sign_up_view.xml:15]: don't nest relative layouts

**Review by JPrez38 — CHANGES_REQUESTED** [review]
Some xml changes mostly. Looks pretty good.

File: app/src/main/java/preston/com/stockapp/HomeActivity.java (hunk 1 above)
Comment (Calvin906) [reply app/src/main/java/preston/com/stockapp/HomeActivity.java]: Done

File: app/src/main/java/preston/com/stockapp/HomeActivity.java (hunk 2 above)
Comment (Calvin906) [reply app/src/main/java/preston/com/stockapp/HomeActivity.java]: Done

File: app/src/main/java/preston/com/stockapp/util/LoginFragment.java (hunk 3 above)
Comment (Calvin906) [reply app/src/main/java/preston/com/stockapp/util/LoginFragment.java]: Done

File: app/src/main/java/preston/com/stockapp/util/SignUpFragment.java (hunk 4 above)
Comment (Calvin906) [reply app/src/main/java/preston/com/stockapp/util/SignUpFragment.java]: Done

File: app/src/main/java/preston/com/stockapp/util/SignUpFragment.java (hunk 5 above)
Comment (Calvin906) [reply app/src/main/java/preston/com/stockapp/util/SignUpFragment.java]: Done

File: app/src/main/res/layout/home_activity.xml (hunk 6 above)
Comment (Calvin906) [reply app/src/main/res/layout/home_activity.xml:15]: Done

File: app/src/main/res/layout/login_view.xml (hunk 7 above)
Comment (Calvin906) [reply app/src/main/res/layout/login_view.xml:15]: Done

File: app/src/main/res/layout/login_view.xml (hunk 8 above)
Comment (Calvin906) [reply app/src/main/res/layout/login_view.xml:47]: Done

**Pull request merged**